src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5949,8 +5956,41 @@ private bool IsStartOfTypeParameter()
                 return false;
 
             // possible attributes
-            if (this.CurrentToken.Kind == SyntaxKind.OpenBracketToken && this.PeekToken(1).Kind != SyntaxKind.CloseBracketToken)
-                return true;
+            if (this.CurrentToken.Kind == SyntaxKind.OpenBracketToken)
```

Comment:
ok.  i was actually able to simplify this to just:

```c#
            // possible attributes.
            // Variance.
            if (this.CurrentToken.Kind is SyntaxKind.OpenBracketToken or SyntaxKind.InKeyword or SyntaxKind.OutKeyword)
                return true;
```

and i got no regressions (in the syntax tests at least).  i'm going to go with that much simpler approach as i don't really see a reason why we would accept `[ <non-close-brace` as an attribute, but not `[]` as well.